common/src/main/java/org/conscrypt/Conscrypt.java

```diff
@@ -705,4 +708,59 @@ public static String getApplicationProtocol(SSLEngine engine) {
             int length) throws SSLException {
         return toConscrypt(engine).exportKeyingMaterial(label, context, length);
     }
+
+    /**
+     * Indicates whether the given {@link TrustManager} was created by this distribution of
+     * Conscrypt.
+     */
+    public static boolean isConscrypt(TrustManager trustManager) {
+        return trustManager instanceof TrustManagerImpl;
+    }
+
+    private static TrustManagerImpl toConscrypt(TrustManager trustManager) {
+        if (!isConscrypt(trustManager)) {
+            throw new IllegalArgumentException(
+                "Not a conscrypt trust manager: " + trustManager.getClass().getName());
+        }
+        return (TrustManagerImpl) trustManager;
+    }
+
+    /**
+     * Set the default hostname verifier that will be used for HTTPS endpoint identification by
+     * Conscrypt trust managers.  If {@code null} (the default), endpoint identification will use
+     * the default hostname verifier set in
+     * {@link HttpsURLConnection#setDefaultHostnameVerifier(HostnameVerifier)}.
+     */
+    public synchronized static void setDefaultHostnameVerifier(HostnameVerifier verifier) {
+        TrustManagerImpl.setDefaultHostnameVerifier(verifier);
+    }
+
+    /**
+     * Returns the currently-set default hostname verifier for Conscrypt trust managers.
+     *
+     * @see #setDefaultHostnameVerifier(HostnameVerifier)
+     */
+    public synchronized static HostnameVerifier getDefaultHostnameVerifier(TrustManager trustManager) {
+        return TrustManagerImpl.getDefaultHostnameVerifier();
+    }
+
+    /**
+     * Set the hostname verifier that will be used for HTTPS endpoint identification by the
+     * given trust manager.  If {@code null} (the default), endpoint identification will use the
+     * default hostname verifier set in {@link #setDefaultHostnameVerifier(HostnameVerifier)}.
+     */
+    public static void setHostnameVerifier(TrustManager trustManager, HostnameVerifier verifier) {
```

Comment:
Throws IllegalArgumentException if trustManager is not the expected type. Worth documenting?

Reply:
Good call, done.